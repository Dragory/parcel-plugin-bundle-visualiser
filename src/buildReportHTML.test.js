const buildReportHTML = require('./buildReportHTML');

jest.mock('fs'); //so that entire foamtree script isnt loaded


describe('./src/buildReportHTML', ()=>{

  it('should generate a report', ()=>{
    const mockTreeData = {
      groups:[
        { label:'thing1', weight:10 },
        { label:'thing2', weight:20 }
      ]
    };
    const report = buildReportHTML(mockTreeData);
    expect(report).toMatchSnapshot();
  });

});