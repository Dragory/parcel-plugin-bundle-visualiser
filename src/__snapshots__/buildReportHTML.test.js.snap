// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./src/buildReportHTML should generate a report 1`] = `
"
  <!DOCTYPE html>
  <html>
    <head>
      <title>FoamTree Quick Start</title>
      <meta charset=\\"utf-8\\" />
    </head>
  
    <body>

      <div class=\\"container\\">
        <div id=\\"visualization\\"></div>
      </div>
      <div class=\\"tooltip\\">This is a test</div>

      <style>
        body{ 
          margin: 0; 
          overflow: hidden;
          font-family: sans-serif;
        }
        .container{ 
          display: flex; 
          height: 100vh; 
        }
        #visualization{ 
          width: 100%; 
        }
        .tooltip {
            position: absolute;
            z-index: 1000;
            max-width: 320px;
            word-break: break-all;
            display: none;
            background: #ddd;
            opacity: 0.85;
            padding: 10px;
        }
        .tooltip em{
          display: inline-block;
          font-style: normal;
          font-weight: bold;
          margin-bottom: 10px;
        }
      </style>

      <script>undefined</script>
      <script>
      (function(){

        var foamtree;

        window.addEventListener('load', function() {
          foamtree = new CarrotSearchFoamTree({
            id: 'visualization',
            dataObject: {\\"groups\\":[{\\"label\\":\\"thing1\\",\\"weight\\":10},{\\"label\\":\\"thing2\\",\\"weight\\":20}]},
            layoutByWeightOrder:true,
            layout: 'squarified',
            stacking: 'flattened',
            pixelRatio: window.devicePixelRatio || 1,
            maxGroupLevelsDrawn: Number.MAX_VALUE,
            maxGroupLevelsAttached:Number.MAX_VALUE,
            maxGroupLabelLevelsDrawn: Number.MAX_VALUE,
            rolloutDuration: 0,
            pullbackDuration: 0,
            fadeDuration: 0,
            zoomMouseWheelDuration: 300,
            openCloseDuration: 200,
            groupLabelVerticalPadding: 0.2,
            groupBorderRadius: 0,
            
            //TODO: follow up later
            onGroupHover:function(e){
              if (e.group && e.group.label) {
                console.log(e.group);
                tooltip.style.display = \\"block\\";
                tooltip.innerHTML = formatTooltip(e.group);
              } else {
                tooltip.style.display = \\"none\\";
              }
            },

            //zoom to group rather than that weird pop out thing
            onGroupDoubleClick: function(e) {
              e.preventDefault();
              var group = e.group;
              var toZoom;
              if (group) {
                toZoom = e.secondary ? group.parent : group;
              } else {
                toZoom = this.get(\\"dataObject\\");
              }
              this.zoom(toZoom);
            }

          });
        });
  
        window.addEventListener(\\"resize\\", (function() {
          var timeout;
          return function() {
            window.clearTimeout(timeout);
            timeout = window.setTimeout(foamtree.resize, 300);
          };
        })());

        function formatTooltip(group){
          var gzipSize = group.formattedGzipSize ? (' (gzipped: ' + group.formattedGzipSize +  ')') : '';
          var t = '<em>' + group.label + '</em>';
          t += '<div>' + group.path + '</div>';
          t += '<div>Size:' + group.formattedSize + gzipSize + '</div>';
          t += '<div>Time: ' + group.formattedTime + '</div>';
          return t;
        }

        var tooltip = document.querySelector('.tooltip');
        document.addEventListener('mousemove', function(e){
          tooltip.style.left =
              e.pageX + tooltip.clientWidth + 20 < document.body.clientWidth ?
              e.pageX + 10 + 'px' :
              document.body.clientWidth - 10 - tooltip.clientWidth + 'px';
          tooltip.style.top =
              e.pageY + tooltip.clientHeight + 20 <
              document.body.clientHeight ?
              e.pageY + 10 + 'px' : 
              document.body.clientHeight - 10 - tooltip.clientHeight + 'px';
        });

      })();
      </script>
    </body>
  </html>
  "
`;
